### PR TITLE
msgpack-cxx: change cmake config file name

### PR DESCRIPTION
**Pull request by uyha**
Upstream README is wrong, a PR is already created there to fix it.

### Summary
Changes to recipe:  **lib/[version]**

#### Details

---
- [ ] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [ ] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [ ] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by AbrilRBS — COMMENTED** [review]
Thanks @uyha! I'll need to double-check with the team what the best approach to avoid breaking potential users of the old buggy name would be (Probably a `cmake_target_alias` property could be set!), thanks for your patience :)

**Review by AbrilRBS — APPROVED** [review]
@uyha sorry for the delay, this was trickier than expected to get right.

The final approach will be to keep the old buggy name for old published versions, and only from a future new version onwards to fix the name, as not to break current consumers with the new revision.

Thanks a lot for taking the time to create the PR in the first place, we appreciate it :)

**Comment by AbrilRBS** [comment]
@uyha do you plan on opening a PR upstream to fix their readme?

**Review by ErniGH — APPROVED** [review]
Thanks! 😄

File: recipes/msgpack-cxx/all/conanfile.py [hunk 1]
```diff
@@ -24,15 +24,16 @@ class MsgpackCXXConan(ConanFile):
     }
     no_copy_source = True
 
-    def configure_options(self):
-        # No boost was added since 4.1.0
+    def config_options(self):
```
Comment (AbrilRBS) [inline recipes/msgpack-cxx/all/conanfile.py:27]: The option was not being removed because the method was not named correctly!

**Review by AbrilRBS — APPROVED** [review]

**Review by ErniGH — APPROVED** [review]

**Comment by uyha** [comment]
> @uyha do you plan on opening a PR upstream to fix their readme?

I did, https://github.com/msgpack/msgpack-c/pull/1142

**Comment by uyha** [comment]
> @uyha sorry for the delay, this was trickier than expected to get right.
> 
> The final approach will be to keep the old buggy name for old published versions, and only from a future new version onwards to fix the name, as not to break current consumers with the new revision.
> 
> Thanks a lot for taking the time to create the PR in the first place, we appreciate it :)

But isn't introducing a mechanism for aliasing better? it introduces an opporunity to fix buggy situations like this.

**Pull request merged**